src/components/DashboardIframe/DashboardIframe.tsx

```diff
@@ -32,23 +33,73 @@ const DashboardIframeCard = ({ dashboard }: { dashboard: {url: string} }) => {
           width="100%"
           height={800}
           frameBorder="0"
+          referrerPolicy="strict-origin-when-cross-origin"
         />
       </CardContent>
     </Card>
-  );
+  )
 };
 
 export const DashboardIframe = (props: {uuid: string}) => {
+  // Add event listener to indicate that we're on the login page (no login cookies). Because we don't
+  // want to actually support the login page in an iframe, which would be necessary as part of the login
+  // redirect flow, instead we direct users to open a new tab and log into Okay. This will set their
+  // browser cookies and authenticate them for the iframe request.
+  const config = useApi(configApiRef);
+  const domain = config.getOptionalString('okay.iframe.domain');
+
+  const [login, setLogin] = React.useState({ 
+    renderLogin: false,
+    refreshPage: false
+  })
+
+  window.addEventListener('message', (event) => {
+    if (event.origin === domain && event.type === 'message' && event.data === 'login_page') {
+      setLogin({ renderLogin: true, refreshPage: false })
+    } else {
+      // postMessage sent from different origin than the iframe, do nothing
+    }
+  }, false);
+
+  
   const okayApi = useApi(okayBackendPluginApiRef);
-const { value, loading, error } = useAsync(async () => await okayApi.iframeUrl(props.uuid));
+  const { value, loading, error } = useAsync(async () => await okayApi.iframeUrl(props.uuid));
 
   if (loading) {
     return <Progress />;
   } else if (error || !value) {
     return <Alert severity="error">{error ? error.message : 'Could not fetch iframe URL'}</Alert>;
+  } else if (login.renderLogin) {
+    return (
+      <Card>
+        <CardContent>
+          <Typography gutterBottom variant="h5" component="div">
+            Okay Dashboard
+          </Typography>
+          <Typography gutterBottom variant="body2" color="textSecondary">
+            Log in to Okay to see your dashboard
+          </Typography>
+          <Button size="small" variant="contained" onClick={() => {
+              setLogin({ renderLogin: false, refreshPage: true });
+              window.open(`${domain}`, '_blank');
+          }}>Take me to Okay</Button>
+        </CardContent>
+      </Card>
+    )
+  } else if (login.refreshPage) {
```

Comment:
nit: just curious, will users get this message if they do not successfully login?

Reply:
yes, wasn't sure of a better way to do this -- we could auto-refresh but I didn't really want to do that in case the user never logged in for whatever reason